core/shared/src/main/scala/org/typelevel/log4cats/package.scala

```diff
@@ -0,0 +1,123 @@
+   *
+   * @tparam F[_]
+   *  the effect type in which logging is done.
+   */
+  @scala.annotation.implicitNotFound(
+    """
+Implicit not found for Logging[${F}]. Which is a convenience type alias.
+
+Keep in mind that:
+
+a) Logging[${F}] has to be created _explicitely_ once (at least) in your code
+
+b) for the type Logging, ${F} is _only_ the effect in which logging is done.
+    ```
+    Logging[${F}].create.info(message) : ${F}[Unit]
+    ```
+
+c) The Logging[${F}] type itself described logging creation as being pure, i.e. in cats.Id
+
+d) Logging[${F}] is defined as creating _only_ loggers of type SelfAwareStructuredLogger[${F}].
+
+The full definition of Logging is:
+    ```
+    type Logging[F[_]] = GenLogging[cats.Id, SelfAwareStructuredLogger[F]]
+    ```
+
+Therefore, your problem might be:
+    1) you didn't create a Logging[${F}] to begin with, or didn't mark it as implicit
+    2) you only have a GenLogger[G[_], SelfAwareStructuredLogger[${F}]] for some G[_] type other than cats.Id
+    3) you do actually have a GenLogger[Id, L], but where L is not SelfAwareStructuredLogger[${F}].
+
+If you are unsure how to create a Logging[${F}], then you can to look at the `log4cats-slf4j`, 
+or `log4cats-noop` modules for concrete implementations.
+
+Example for slf4j:
+    ```
+    import cats.effect.IO
+
+    // we create our Logging[IO]
+    implicit val logging: Logging[IO] = Slf4jLogging.forSync[IO]
+
+    //we summon our instance, and create our logger
+    val logger: SelfAwareStructuredLogger[IO] = Logging[IO].create
+    logger.info("logging in IO!"): IO[Unit]
+    ```
+"""
+  )
+  type Logging[F[_]] = GenLogging[cats.Id, SelfAwareStructuredLogger[F]]
+
+  /**
+   * Convenience type
+   *
+   * @tparam F[_]
+   *  the effect type of both logger creation, and logging.
+   */
+  @scala.annotation.implicitNotFound(
+    """
+Implicit not found for LoggingF[${F}]. Which is a convenience type alias.
+
+Keep in mind that:
+
+a) LoggingF[${F}] has to be created _explicitely_ once (at least) in your code
+
+b) for the type LoggingF, ${F} is _both_ the effect in which logger creation is done
+   _and_ the effect in which logging is done
+    ```
+    val loggerF: ${F}[SelfAwareStructuredLogger] = Logging[${F}].create
+    loggerF.flatMap{logger => logger.info("the result of the .info is in F"): ${F}[Unit] }: ${F}[Unit]
+    ```
+
+c) LoggingF[${F}] is defined as creating _only_ loggers of type SelfAwareStructuredLogger[${F}].
+
+The full definition of Logging is:
+    ```
+    type LoggingF[F[_]] = GenLogging[F, SelfAwareStructuredLogger[F]]
+    ```
+Therefore, your problem might be:
+    1) you didn't create a LoggingF[${F}] to begin with, or didn't mark it as implicit
+    2) you only have a GenLoggerF[G[_], SelfAwareStructuredLogger[${F}]] for some G[_] type other than type ${F}
+    3) you do actually have a GenLogger[${F}, L], but where L is not SelfAwareStructuredLogger[${F}].
```

Comment:
Can/should we support users doing `import org.typelevel.log4s.Slf4jLogging.implicits._`?

Reply:
I think I would prefer not to 😅

It would carry something like `implicit def genericSlf4jLogging[F[_]: Sync]: Slf4jLogging[F, F]`? And others, provided they don't wind up conflicting with each other under the same import.

Reply:
Yeah, why not? Shouldn't conflict as long as types are different, so it could be both `Logging` and `LoggingF` under one object. I've no objections to not having it as an experienced Scala-FP-CE-TF developer but not everybody's like that, and a page's worth of text for `implicitNotFound` to describe how to refactor all your code is probably going to alienate more than help.

Reply:
Good points. But I am still in favor of a page worth of `implicitNotFound` because adopters of Logging as capability are probably experienced Scala-FP-CE-TF devs, since it's an entirely new feature.

I can also move the practical examples first, with the alternative of a-la-carte `implicits` import as well, and then move the general points about what might be wrong at the bottom.

Reply:
Yeah that sounds better. Even "experienced Scala-FP-CE-TF devs" prefer a simple entry point (cue: CE2 Timer/ContextShift/Concurrent vs CE3 import unsafe), and for people just trying log4cats out (there's always a lot of people swayed by TF but not yet with enough experience to handle all the issues) that would provide a straightforward experience.

Reply:
Updated the PR, rephrased so it doesn't seem like we're giving refactoring advice. But now it's somehow even longer :sweat_smile: